master/buildbot/test/integration/test_upcloud.py

```diff
@@ -0,0 +1,37 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+from future.utils import string_types
+
+import json
+import os
+from unittest.case import SkipTest
+
+from twisted.internet import defer
+
+from buildbot import util
+from buildbot.config import BuilderConfig
+from buildbot.plugins import schedulers
+from buildbot.plugins import steps
+from buildbot.process.factory import BuildFactory
+from buildbot.process.results import RETRY
+from buildbot.process.results import SUCCESS
+from buildbot.test.fake.step import BuildStepController
+from buildbot.test.util.integration import RunMasterBase
+from buildbot.worker.upcloud import UpcloudLatentWorker
+
+## FIXME: How to make mock HTTP requests?!?
```

Comment:
using FakeHttpClientService. There are plenty of examples and doc is here:
http://docs.buildbot.net/current/developer/utils.html#buildbot-test-fake-httpclientservice